LedgerHQ/ledger-live-desktop
# LL-1931 [Mac] Notarize .app and .dmg files for Catalina compatibility

MacOS Catalina in its default configuration refuses to open apps unless they have been sent to Apple for notarization.
This PR adds that into our build system.

This also enables [Hardened Runtime](https://developer.apple.com/documentation/security/hardened_runtime_entitlements), as it's a requirement for notarization.

Since we're still stuck on an old `electron-builder` version for now, I had to fork it (only `electron-updater` was forked up until now) and backport some needed features and fixes (see https://github.com/LedgerHQ/electron-builder)

### Type

Bug Fix, Feature

### Context

https://ledgerhq.atlassian.net/browse/LL-1931

### Parts of the app affected

App building
Runtime

### Test plan

- Generate an app specific Apple ID password on [appleid.apple.com](https://appleid.apple.com)
- In your `.env` file, add `APPLEID` and `APPLEID_PASSWORD` with your Apple ID login and the password you just generated, respectively
- Run `yarn dist:signed`
- Open the resulting dmg file on macOS Catalina, and run the app from it, it should launch without complaining.

### Potential regressions checks

- Make sure the app still builds on Windows and Linux
- [Mac only] Make sure _hardened runtime_ didn't brake anything (components most likely to be affected are libcore and device communication)
